modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java

```diff
@@ -328,7 +328,20 @@ public String getTypeDeclaration(Schema p) {
         if (ModelUtils.isArraySchema(p)) {
             ArraySchema ap = (ArraySchema) p;
             Schema inner = ap.getItems();
-            return "[]" + getTypeDeclaration(ModelUtils.unaliasSchema(this.openAPI, inner));
+            // In OAS 3.0.x, the array "items" attribute is required.
+            // In OAS >= 3.1, the array "items" attribute is optional such that the OAS
+            // specification is aligned with the JSON schema specification.
+            // When "items" is not specified, the elements of the array may be anything at all.
+            if (inner != null) {
+                inner = ModelUtils.unaliasSchema(this.openAPI, inner);
+            }
+            String typDecl;
+            if (inner != null) {
+                typDecl = getTypeDeclaration(inner);
+            } else {
+                typDecl = "interface{}";
```

Comment:
I may submit a PR later to avoid hardcoding "interface{}" and use the value defined in typeMapping instead.